src/tribol/search/InterfacePairFinder.cpp

```diff
@@ -610,27 +612,27 @@ class BvhSearch : public SearchBase {
         } );
   }  // end findInterfacePairs()
 
-  void buildMeshBBoxes( ArrayT<BoxT>& boxes, const MeshData::Viewer& mesh )
+  void buildMeshBBoxes( ArrayT<BoxT>& boxes, const MeshData::Viewer& mesh, RealT binning_proximity )
   {
     auto boxes1_view = boxes.view();
     forAllExec( m_coupling_scheme->getExecutionMode(), mesh.numberOfElements(),
-                [this, mesh, boxes1_view] TRIBOL_HOST_DEVICE( IndexT i ) {
+                [this, mesh, boxes1_view, binning_proximity] TRIBOL_HOST_DEVICE( IndexT i ) {
                   BoxT box;
                   auto num_nodes_per_elem = mesh.numberOfNodesPerElement();
                   for ( IndexT j{ 0 }; j < num_nodes_per_elem; ++j ) {
                     IndexT node_id = mesh.getGlobalNodeId( i, j );
-                    RealT pos[3];
-                    pos[0] = mesh.getPosition()[0][node_id];
-                    pos[1] = mesh.getPosition()[1][node_id];
-                    pos[2] = mesh.getPosition()[2][node_id];  // unused if D==2
-                    box.addPoint( PointT( pos ) );
+                    PointT pos;
+                    for ( int d{ 0 }; d < D; ++d ) {
+                      pos[d] = mesh.getPosition()[d][node_id];
+                    }
+                    box.addPoint( pos );
                   }
                   // Expand the bounding box in the face normal direction
                   RealT vnorm[3];
                   mesh.getFaceNormal( i, vnorm );
                   VectorT faceNormal( vnorm );
                   RealT faceRadius = mesh.getFaceRadius()[i];
-                  expandBBoxNormal( box, faceNormal, faceRadius );
+                  expandBBoxNormal( box, faceNormal, binning_proximity * faceRadius );
```

Comment:
I see that faceRadius was always used here. Perhaps this is still what we want, but we also should have the meshes element thicknesses associated with the faces. Would that be more appropriate? At the time we only had the face radius as a proximity length scale. The potential issue here is that for thin but long elements, the face radius will be more inclusive for adjacent and nearly adjacent elements. For really small faces but deep elements, the element thickness will be more inclusive. Thoughts?

Reply:
I think face radius is fine; there isn't always a notion of thickness at the Tribol mesh level (unless the field is registered with Tribol).